### DISCUSSION TITLE
ListArray collection V8Array problem

### DISCUSSION THREAD

**Discussion by nicloay** [ClearFoundry/ClearScript#501] [Q&A] [answered]
I have the following method at C# side
```
public static void logArray(object texts)
{   
    Debug.Log("Object = "+ texts.GetType());
}
```
and then from javascript, I call something like this
```
console.logArray(["hello", "world"])
```
and then I receive the following output in the console
```
[23:39:43:849] Object = Microsoft.ClearScript.V8.V8ScriptItem+V8Array
```

The problem is that I want to have logArray method array of string arguments. and not V8Array. is it possible to set it up somehow?
Is it the only option as described [here](https://github.com/microsoft/ClearScript/issues/178)
use IList? 
```
public static void logArray(IList texts)
{   
    Debug.Log("Object = "+ texts.GetType());
}
```

**Comment by ClearScriptLib** [comment, marked as answer]
Hi @nicloay,

As you noted, JavaScript arrays implement `IList` and `IList<object>` when accessed from .NET. That makes it very easy to convert them to .NET arrays, lists, etc. Is that not sufficient for your scenario? Please clarify.

Thanks!

**Reply by nicloay** [reply]
Yes, that makes sense to me. and I can cast it in .Net, but may I also ask you how to return an array of data from .Net to JS e.g. ```['one','two']``` ?
I can't control the js script, it's come from the backend. so for me would be better to form a supported structure on the .Net side, which will be translated correctly to the array on the js side. 
thanks.

**Reply by nicloay** [reply]
oops. sorry I see you already answered this question [here](https://github.com/microsoft/ClearScript/issues/375).
So here is my working example. The method receives messages from javascript and then return collected messages from .Net back to JS
```
public async Task<object> sendMessages(IList messages)
{
    foreach (var message in messages)
    {
        var data = JsonConvert.DeserializeObject<IMessage>((string)message, _converter);
        MessageHandler.HandleMessage(data);
    }

    return _sandbox.Script.Array.from(_messageQueue.PopMessagesAsStringList());
}
```
